### PR TITLE
Add blog post - SCS container registry migration and upgrade

### PR DESCRIPTION
**Pull request by matofeder**
Note: failed build CI pipeline should pass after https://github.com/SovereignCloudStack/website/pull/666 is merged

### CODE REVIEW

File: _i18n/en/_posts/blog/2023-05-30-registry-migration-upgrade.md [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+and "[upgrade-and-migrate](https://goharbor.io/docs/main/administration/upgrade/)" 
+with up-to-date commands, explanations, and detailed descriptions of prerequisites needed
+for this maintenance work.
+
+# Plan
+
+Originally, the SCS Harbor container registry instance (https://registry.scs.community/)
+was operated on SCS infrastructure in Kubernetes cluster v1.18.20. This Harbor instance
+with version 2.2.4 was deployed by [harbor-helm](https://github.com/goharbor/harbor-helm)
+chart v1.6.0. This version of  harbor-helm chart is affected by [security issue](https://github.com/goharbor/harbor/security/advisories/GHSA-j7jh-fmcm-xxwv)
```
Comment (chess-knight) [inline _i18n/en/_posts/blog/2023-05-30-registry-migration-upgrade.md:31]:
```suggestion
chart v1.6.0. This version of harbor-helm chart is affected by [security issue](https://github.com/goharbor/harbor/security/advisories/GHSA-j7jh-fmcm-xxwv)
```

File: _i18n/en/_posts/blog/2023-05-30-registry-migration-upgrade.md [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+In addition, Harbor deployment didn't use the ability of Harbor to operate in the high 
+availability mode, which might be beneficial for many reasons.
+
+Our plan was to migrate this Harbor instance to the up-to-date SCS Kubernetes cluster
+with version 1.25.6 which has been spawned on SCS-compatible infrastructure, but operated
+by a different cloud service provider. After the migration process, the upgrade to the
+latest stable version of the Harbor container registry v2.7.1 using the latest stable
+version of harbor-helm chart v1.11.1 (that is not affected by mentioned security issue)
+was planned. The upgrade procedure contains also increasing the number of replicas for
+Harbor fundamental services to >=2 to achieve the HA of Harbor components. The HA mode
```
Comment (chess-knight) [inline _i18n/en/_posts/blog/2023-05-30-registry-migration-upgrade.md:46]: Maybe add a link to https://goharbor.io/docs/main/install-config/harbor-ha-helm/ for reference.

File: _i18n/en/_posts/blog/2023-05-30-registry-migration-upgrade.md [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+the Harbor database as a storage backend for job service instead of PVs. As the RWX 
+access mode for PVs is not currently supported the second option has been planned. 
+In the case of the registry service, the administrator can again configure shared PVs 
+for all pods, or alternatively, external object storage could be used as a registry 
+storage backend. Our first choice was to use object store as a backend for image blobs 
+as the amount of data here is expected to be large and the object store drive is a good
+option when you want to store this type of unstructured data at scale.
+See the original and planned architecture high-level diagram below:
+
+![scs_harbor_migration](images/scs_harbor_migration.png)
```
Comment (chess-knight) [inline _i18n/en/_posts/blog/2023-05-30-registry-migration-upgrade.md:65]: This should be probably removed.

File: _i18n/en/_posts/blog/2023-05-30-registry-migration-upgrade.md [hunk 4]
```diff
@@ -0,0 +1,376 @@
...
+          mountPath: /storage
+EOF
+
+kubectl --kubeconfig <path of Cluster_B kubeconfig> apply -f helper-pod.yaml
+```
+
+Using this pod we copied all image blobs from PV to the object store.
+
+```bash
+# create `~/.aws/credentials`file with the right secrets
```
Comment (chess-knight) [inline _i18n/en/_posts/blog/2023-05-30-registry-migration-upgrade.md:359]: Maybe add `kubectl exec` command before that.

**Review by chess-knight — COMMENTED** [review]
Overall LGTM!
Just small typos/suggestions.

File: _i18n/en/_posts/blog/2023-05-30-registry-migration-upgrade.md (hunk 4 above)
Comment (matofeder) [reply _i18n/en/_posts/blog/2023-05-30-registry-migration-upgrade.md:357]: addressed in a3e5f37ce208591c4fb463b35f066e76557bbec5

File: _i18n/en/_posts/blog/2023-05-30-registry-migration-upgrade.md (hunk 3 above)
Comment (matofeder) [reply _i18n/en/_posts/blog/2023-05-30-registry-migration-upgrade.md]: addressed in a3e5f37ce208591c4fb463b35f066e76557bbec5

File: _i18n/en/_posts/blog/2023-05-30-registry-migration-upgrade.md (hunk 2 above)
Comment (matofeder) [reply _i18n/en/_posts/blog/2023-05-30-registry-migration-upgrade.md]: addressed in a3e5f37ce208591c4fb463b35f066e76557bbec5

**Comment by fkr** [comment]
paging @ajfriesen

**Review by ajfriesen — APPROVED** [review]
I have added a commit with minor grammar and sentence structure changes.

Other than that: Great post, I could follow it even without context 👍🏻

**Comment by fkr** [comment]
> I have added a commit with minor grammar and sentence structure changes.
> 
> Other than that: Great post, I could follow it even without context 👍🏻

thanks @ajfriesen for taking the time! :)

**Pull request merged**